src/AlternateLife.RageMP.Net/Scripting/ScriptingClasses/Commands.cs

```diff
@@ -1,157 +1,207 @@
+                bool isParams = IsParams(expectedParameters[i]);
 
-                _commands.TryRemove(command.Key, out _);
+                if (i == expectedParameters.Count - 1 && isParams)
+                {
+                    var paramList = new List<object>();
+                    for (; i < args.Count; i++)
+                    {
+                        paramList.Add(args[i]);
+                    }
+
+                    parsedArgs.Add(paramList.ToArray());
+                    break;
+                }
+
+                if (i == expectedParameters.Count - 1 && !isParams && args.Count > expectedParameters.Count)
+                {
+                    return null;
+                }
+                
+                if (!_typeParsingSwitch.ContainsKey(expectedType))
+                {
+                    return null;
+                }
+                
+                if (!_typeParsingSwitch[expectedType].TryParse(args[i], targetType, out var parsedPara))
+                {
+                    return null;
+                }
+                parsedArgs.Add(parsedPara);
             }
+
+            return parsedArgs;
         }
 
-        public async Task ExecuteCommand(IPlayer player, string commandText)
+        public async Task ExecuteCommand(IPlayer player, string text)
         {
-            if (string.IsNullOrWhiteSpace(commandText))
+            string[] commandMessage = text.Split(' ', StringSplitOptions.RemoveEmptyEntries);
+            if (commandMessage.Length == 0)
             {
                 return;
             }
+            string commandname = commandMessage[0];
 
-            var parts = commandText.Trim().Split(' ');
-
-            var commandName = parts[0];
-
-            if (_commands.TryGetValue(commandName, out var commandInformation) == false)
+            var command = _commands.FirstOrDefault(c => c.Name == commandname);
+            if (command == null)
             {
+                await player.OutputChatBoxAsync("Command not found.");
                 return;
             }
 
-            try
-            {
-                await commandInformation.Callback(player, parts.Skip(1).Where(x => string.IsNullOrWhiteSpace(x) == false).ToArray());
-            }
-            catch (Exception e)
+            string[] commandArgs = commandMessage.Skip(1).Where(s => !string.IsNullOrWhiteSpace(s)).ToArray();
+            var parsedArgs = ProcessArguments(commandArgs, command.MethodInfo.GetParameters().Skip(1).ToArray());
+            if (parsedArgs == null)
             {
-                _logger.Error($"An error occured when player {player.Name} executed command: {commandName}: ", e);
+                await player.OutputChatBoxAsync("Type conversion failed. Command parameters are: " + command.GetParameterList());
+                return;
             }
+            
+            object[] invokingArgs = {player};
+            invokingArgs = invokingArgs.AppendArray(parsedArgs.ToArray());
```

Comment:
You already know the amount of elements for this array, why don't you create it with the right size and then fill it with the resulting data? See line 193